examples/angular2/services/store.ts

```diff
@@ -0,0 +1,76 @@
+export class Todo {
+	completed: Boolean;
+	editing: Boolean;
+	title: String;
+	uid: String;
+	cleanTitle(title:String) {
+		return title.trim();
+	}
+	setTitle(title:String) {
+		this.title = this.cleanTitle(title);
+	}
+	constructor(title:String) {
+		this.uid = uuid.v4();
+		this.completed = false;
+		this.editing = false;
+		this.title = this.cleanTitle(title);
+	}
+}
+
+export class TodoStore {
+	todos: Array<Todo>;
+	constructor() {
+		let persistedTodos = store.get('angular2-todos') || [];
+		// Normalize back into classes
+		this.todos = persistedTodos.map( todo => {
+			let ret = new Todo(todo.title);
+			ret.completed = todo.completed;
+			ret.uid = todo.uid;
+			return ret;
+		});
+	}
+	_updateStore() {
+		store.set('angular2-todos', this.todos);
+	}
+	get(state:Object) {
+		return this.todos.filter((todo:Todo) => todo.completed == state.completed);
```

Comment:
Don't know about TS, but in ES6 you don't need the parens with only one argument.